scripts/create_compdb.py

```diff
@@ -66,6 +68,66 @@ def _build_generated_files(
     subprocess.check_call(
         [bazel, "build", "--keep_going", "--remote_download_outputs=toplevel"]
         + generated_file_labels
+        # We also need the Bazel C++ runfiles that aren't "generated", but not
+        # linked into place until built.
+        + ["@bazel_tools//tools/cpp/runfiles:runfiles"]
+    )
+
+
+def _get_config_for_entry(entry: Dict[str, Any]) -> str:
+    """Returns the configuration for a compile command entry."""
+    arguments = entry.get("arguments")
+
+    # Only handle files where the object file argument is easily found as
+    # the last argument, which matches the expected structure from Bazel.
+    if not arguments or len(arguments) < 2 or arguments[-2] != "-o":
+        return "unknown"
+    obj_file = arguments[-1]
+
+    # The configuration is the name of the subdirectory of `bazel-out`.
+    if not obj_file.startswith("bazel-out/"):
+        return "unknown"
+    return str(obj_file.split("/")[1])
```

Comment:
What's the purpose of the `str` on line 90? You seem to be doing string operations on it; if it's for mypy, maybe assert it's a string earlier, e.g.:

```suggestion
    obj_file = arguments[-1]
    assert obj_file is str

    # The configuration is the name of the subdirectory of `bazel-out`.
    if not obj_file.startswith("bazel-out/"):
        return "unknown"
    return obj_file.split("/")[1]
```

Reply:
It was mypy -- and today I learn about that use of assert, cool!